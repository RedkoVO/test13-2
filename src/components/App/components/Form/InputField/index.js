import React from 'react'
import PropTypes from 'prop-types'
import cn from 'classnames'

const InputField = ({ input, id, className, type, name, placeholder, meta: { touched, error } }) => {
  const wpInput = cn(className, { 'input-error': touched && error })

  return (
    <input
      {...input}
      id={id}
      name={name}
      type={type}
      className={wpInput}
      placeholder={placeholder}
    />
  )
}

InputField.propTypes = {
  classNamees: PropTypes.object,
  id: PropTypes.string,
  name: PropTypes.string,
  type: PropTypes.string,
  placeholder: PropTypes.string
}

export default InputField