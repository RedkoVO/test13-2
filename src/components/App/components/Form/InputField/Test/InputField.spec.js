import * as React from 'react'
import renderer from 'react-test-renderer'

import InputField from '../'

it('renders correctly', () => {
  const tree = renderer.create(
    <InputField
      value={'value'}
      id={'id'}
      name={'name'}
      type={'type'}
      className={'wpInput'}
      placeholder={'placeholder'}
      meta={{ touched: true, error: 'error' }}
    />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})