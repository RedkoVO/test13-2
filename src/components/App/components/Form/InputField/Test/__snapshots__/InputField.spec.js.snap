// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<input
  className="wpInput input-error"
  id="id"
  name="name"
  placeholder="placeholder"
  type="type"
/>
`;
